### PR TITLE
docs: Improve the README

### LINKED ISSUE #111: Improve the README
The `README` is a bit *too* minimal.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add in details to the `README`.

Fixes #111.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,137 @@
...
+[developer guide](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md).
+
+## Configuration
+
+The runtime uses a TOML format configuration file called `configuration.toml`.
+The file contains comments explaining all options.
+
+> **Note:**
+> 
+> Most users will not need to modify the default configuration.
```
Comment (grahamwhaley) [inline README.md]: I'm not sure how true this is - maybe we can rephrase it more like:
`The defaults provide a good default configuration. You may only need to modify this file if you have specialist advanced user needs`
or similar (that still needs work...)

File: README.md [hunk 2]
```diff
@@ -1,5 +1,137 @@
...
+
+The command below will list the full paths to the configuration files the runtime
+will attempt to load. The first path that exists will be used:
+
+```bash
+$ kata-runtime --kata-show-default-config-paths
+```
+
+Aside from the built-in locations, it is possible to
+specify the path to a custom configuration file using the `` option:
```
Comment (grahamwhaley) [inline README.md]: the `` option - seems to have something missing between the quotes?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,137 @@
...
+the configuration file being used), run:
+
+```bash
+$ kata-runtime kata-env
+```
+
+## Logging
+
+The runtime provides `--log=` and `--log-format=` options. However, the
+runtime will always log to the system log (syslog or `journald`).
```
Comment (grahamwhaley) [inline README.md]: `` around syslog?

File: README.md [hunk 4]
```diff
@@ -1,5 +1,137 @@
...
+## Debugging
+
+See the
+[debugging section of the developer guide](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#enable-full-debug).
+
+## Community
+
+See [the community repository](https://github.com/kata-containers/community).
+
+## Home Page
```
Comment (grahamwhaley) [inline README.md]: I would maybe put this at the top, not the bottom?

File: README.md (hunk 2 above)
Comment (jodh-intel) [reply README.md]: arf. updated.

File: README.md (hunk 3 above)
Comment (jodh-intel) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (jodh-intel) [reply README.md]: Reworked to remove that section entirely and put the links at the top.

**Comment by devimc** [comment]
lgtm

**Comment by jodh-intel** [comment]
Hi @klynnrif - ptal.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,137 @@
...
+    * [Hardware requirements](#hardware-requirements)
+* [Quick start for developers](#quick-start-for-developers)
+* [Configuration](#configuration)
+* [Logging](#logging)
+* [Debugging](#debugging)
+* [Community](#community)
+
+## Introduction
+
+`kata-runtime`, henceforth known simply as "the runtime", is the
```
Comment (klynnrif) [inline README.md]: `kata-runtime`, referred to as "the runtime", is the

File: README.md [hunk 6]
```diff
@@ -1,5 +1,137 @@
...
+CLI (command-line interface) part of the Kata Containers runtime component. It leverages the
```
Comment (klynnrif) [inline README.md]: Command-line interface (CLI) part of the Kata Containers runtime component.

File: README.md [hunk 7]
```diff
@@ -1,5 +1,137 @@
...
+The runtime has a built-in command to determine if your host system is capable
+of running a Kata Container:
+
+```bash
+$ kata-runtime kata-check
+```
+
+> **Note:**
+>
+> If you run the command above as the `root` user, further checks will be
```
Comment (klynnrif) [inline README.md]: > If you run the previous command as the `root` user, further checks will be

File: README.md [hunk 8]
```diff
@@ -1,5 +1,137 @@
...
+
+## Configuration
+
+The runtime uses a TOML format configuration file called `configuration.toml`.
+The file contains comments explaining all options.
+
+> **Note:**
+>
+> The initial values in the configuration file provide a good default configuration.
+> You may only need to modify this file if you have specialist needs.
```
Comment (klynnrif) [inline README.md]: > You might need to modify this file if you have specialist needs.

File: README.md [hunk 9]
```diff
@@ -1,5 +1,137 @@
...
+
+Since the runtime supports a
+[stateless system](https://clearlinux.org/features/stateless),
+it checks for this configuration file in multiple locations, two of which are
+built in to the runtime. The default location is
+`/usr/share/defaults/kata-containers/configuration.toml` for a standard
+system. However, if `/etc/kata-containers/configuration.toml` exists, this
+will take priority.
```
Comment (klynnrif) [inline README.md]: takes priority.

File: README.md [hunk 10]
```diff
@@ -1,5 +1,137 @@
...
+
+The command below will list the full paths to the configuration files the runtime
```
Comment (klynnrif) [inline README.md]: Lines 91-92: The command below lists the full paths to the configuration files that the runtime attempts to load. The first path that exists is used:

File: README.md [hunk 11]
```diff
@@ -1,5 +1,137 @@
...
+the configuration file being used), run:
+
+```bash
+$ kata-runtime kata-env
+```
+
+## Logging
+
+The runtime provides `--log=` and `--log-format=` options. However, the
+runtime will always log to the system log (`syslog` or `journald`).
```
Comment (klynnrif) [inline README.md]: runtime always logs to the system log (`syslog` or `journald`).

File: README.md [hunk 12]
```diff
@@ -1,5 +1,137 @@
...
+
+To view runtime log output:
+
+```bash
+$ sudo journalctl -t kata-runtime
+```
+
+For more thorough analysis and details of how to obtain logs for other system components, see the documentation for the
```
Comment (klynnrif) [inline README.md]: For detailed information and analysis on obtaining logs for other system components, see the documentation for the...

**Comment by jodh-intel** [comment]
Thanks @klynnrif - branch updated.

**Comment by jodh-intel** [comment]
CI is failing as https://github.com/kata-containers/tests/pull/181 isn't merged yet.

**Comment by egernst** [comment]
Restarted travis -- will merge if it passes.

**Pull request merged**